Code/Tools/AssetProcessor/native/AssetManager/assetProcessorManager.cpp

```diff
@@ -1481,15 +1485,26 @@ namespace AssetProcessor
                     message.m_legacyAssetIds.push_back(legacyAssetId);
                 }
 
-                if (legacySourceAssetId != assetId)
+                SourceAssetReference sourceAsset(source.m_scanFolderPK, source.m_sourceName.c_str());
+                AZStd::unordered_set<AZ::Data::AssetId> legacySourceAssetIds;
+                auto legacySourceUuids = AssetUtilities::GetLegacySourceUuids(sourceAsset);
+                legacySourceAssetIds.reserve(legacySourceUuids.size());
+
+                for (const auto& legacyUuid : legacySourceUuids)
                 {
-                    message.m_legacyAssetIds.push_back(legacySourceAssetId);
+                    AZ::Data::AssetId legacySourceAssetId(legacyUuid, newProduct.m_subID);
+
+                    if (legacySourceAssetId != assetId)
+                    {
+                        legacySourceAssetIds.emplace(legacySourceAssetId);
+                        message.m_legacyAssetIds.push_back(legacySourceAssetId);
```

Comment:
Should these lines be reversed to avoid a copy?

Reply:
A copy of what?  We're inserting the assetId into the tracking set and the message vector.

Reply:
legacySourceAssetId could be moved to legacySourceAssetIds after a push_back with message, or do we want a "Construct and insert element" here?

Reply:
Ah I see.  Sure, I've updated it to move the assetId into the last container.